packages/core/parcel-bundler/src/builtins/hmr-runtime.js

```diff
@@ -34,16 +36,56 @@ if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
 
     if (data.type === 'update') {
       console.clear();
-
-      data.assets.forEach(function (asset) {
-        hmrApply(global.parcelRequire, asset);
-      });
-
-      data.assets.forEach(function (asset) {
-        if (!asset.isNew) {
-          hmrAccept(global.parcelRequire, asset.id);
-        }
-      });
+      
+      // Handle WebAssembly
+      if (data.assets.some(function(asset) {
+        return ('wasm' in (asset.generated || {}))
+      }) && window.Promise) {
+        Promise.all(data.assets.map(function(asset) {
+          if ('wasm' in (asset.generated || {})) { // Inject wasm into runtime
+            if(window.atob == null) {
+              window.atob = require('atob'); // polyfill for tests
+            }
+            var binStr = window.atob(asset.generated.wasm.blob);
+            var len = binStr.length;
+            var bytes = new Uint8Array(len);
+            for (var i = 0; i < len; ++i)
+              bytes[i] = binStr.charCodeAt(i);
+
+            return WebAssembly.instantiate(bytes).then(function(wasmModule) {
```

Comment:
Synchronous Wasm instantiation and inlining the file content is really not ideal. It's recommend to use `instantiateStreaming`, is there any blocker to use it even for hmr?